src/black/parsing.py

```diff
@@ -92,6 +93,12 @@ def lib2to3_parse(src_txt: str, target_versions: Iterable[TargetVersion] = ()) -
             except IndexError:
                 faulty_line = "<line number missing in source>"
             exc = InvalidInput(f"Cannot parse: {lineno}:{column}: {faulty_line}")
+
+        except TokenError as te:
+            lineno, column = te.args[1]
+            lines = src_txt.splitlines()
+            exc = InvalidInput(f"Cannot parse: {lineno}:{column}: Unexpected EOF")
+
```

Comment:
I included it as a separate block since that seemed more readable to me instead of adding multiple if (isintance,TokenError) in the above block where were handling ParseError

Reply:
Fair enough, I agree the duplication isn't a problem here.